app/src/main/java/org/metabrainz/mobile/api/webservice/MusicBrainzServiceGenerator.java

```diff
@@ -1,21 +1,29 @@
 package org.metabrainz.mobile.api.webservice;
 
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.Interceptor;
 import okhttp3.OkHttpClient;
 import okhttp3.logging.HttpLoggingInterceptor;
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;
 
 public class MusicBrainzServiceGenerator {
-    public static final String API_BASE_URL = "http://metabrainz.org/ws/2/";
-    public static final String AUTH_URL = "https://musicbrainz.org/oauth2/authorize";
-    public static final String CLIENT_ID = "7YOlN2TqOe5LaD1zdZpzeQ";
-    public static final String  CLIENT_SECRET = "iJ45_lhc5-docYBfLNRaaA";
-    public static final String OAUTH_REDIRECT_URI = "http://localhost";
+    public static final String API_BASE_URL = "http://musicbrainz.org/ws/2/";
```

Comment:
I haven't encountered the error yet. I will look further into it. Could you provide a detailed stack trace?

Reply:
### Emulator config at work:

```
Name: Pixel_API_28
CPU/ABI: Google Play Intel Atom (x86)
Path: /Users/miguelherrero/.android/avd/Pixel_API_28.avd
Target: google_apis_playstore [Google Play] (API level 28)
Skin: 1080x1920
SD Card: 512 MB
fastboot.chosenSnapshotFile: 
runtime.network.speed: full
hw.accelerometer: yes
hw.device.name: pixel
hw.lcd.width: 1080
hw.initialOrientation: Portrait
image.androidVersion.api: 28
tag.id: google_apis_playstore
hw.mainKeys: no
hw.camera.front: none
avd.ini.displayname: Pixel API 28
hw.gpu.mode: auto
hw.ramSize: 1536
PlayStore.enabled: true
fastboot.forceColdBoot: no
hw.cpu.ncore: 2
hw.keyboard: yes
hw.sensors.proximity: yes
hw.dPad: no
hw.lcd.height: 1920
vm.heapSize: 228
skin.dynamic: yes
hw.device.manufacturer: Google
hw.gps: yes
skin.path.backup: _no_skin
hw.audioInput: yes
image.sysdir.1: system-images/android-28/google_apis_playstore/x86/
showDeviceFrame: no
hw.camera.back: none
AvdId: Pixel_API_28
hw.lcd.density: 420
hw.arc: false
hw.device.hash2: MD5:55acbc835978f326788ed66a5cd4c9a7
fastboot.forceChosenSnapshotBoot: no
fastboot.forceFastBoot: yes
hw.trackBall: no
hw.battery: yes
hw.sdCard: yes
tag.display: Google Play
runtime.network.latency: none
disk.dataPartition.size: 2G
hw.sensors.orientation: yes
avd.ini.encoding: UTF-8
hw.gpu.enabled: yes
```

### Logcat
```
2019-02-28 12:31:34.154 10150-10150/? I/abrainz.androi: Not late-enabling -Xcheck:jni (already on)
2019-02-28 12:31:34.238 10150-10150/? W/abrainz.androi: Unexpected CPU variant for X86 using defaults: x86
2019-02-28 12:31:34.832 10150-10150/org.metabrainz.android W/abrainz.androi: JIT profile information will not be recorded: profile file does not exits.
2019-02-28 12:31:34.833 10150-10150/org.metabrainz.android I/chatty: uid=10085(org.metabrainz.android) identical 10 lines
2019-02-28 12:31:34.834 10150-10150/org.metabrainz.android W/abrainz.androi: JIT profile information will not be recorded: profile file does not exits.
2019-02-28 12:31:34.880 10150-10150/org.metabrainz.android I/InstantRun: starting instant run server: is main process
2019-02-28 12:31:35.373 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (light greylist, reflection)
2019-02-28 12:31:35.374 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (light greylist, reflection)
2019-02-28 12:31:35.526 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Landroid/widget/AutoCompleteTextView;->doBeforeTextChanged()V (light greylist, reflection)
2019-02-28 12:31:35.527 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Landroid/widget/AutoCompleteTextView;->doAfterTextChanged()V (light greylist, reflection)
2019-02-28 12:31:35.527 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Landroid/widget/AutoCompleteTextView;->ensureImeVisible(Z)V (light greylist, reflection)
2019-02-28 12:31:35.580 10150-10162/org.metabrainz.android I/abrainz.androi: Background concurrent copying GC freed 19925(7MB) AllocSpace objects, 1(20KB) LOS objects, 51% free, 1459KB/2MB, paused 2.557ms total 175.171ms
2019-02-28 12:31:35.970 10150-10150/org.metabrainz.android D/OpenGLRenderer: Skia GL Pipeline
2019-02-28 12:31:36.135 10150-10184/org.metabrainz.android I/ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0
2019-02-28 12:31:36.135 10150-10184/org.metabrainz.android I/ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
2019-02-28 12:31:36.136 10150-10184/org.metabrainz.android I/OpenGLRenderer: Initialized EGL, version 1.4
2019-02-28 12:31:36.136 10150-10184/org.metabrainz.android D/OpenGLRenderer: Swap behavior 1
2019-02-28 12:31:36.138 10150-10184/org.metabrainz.android W/OpenGLRenderer: Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
2019-02-28 12:31:36.138 10150-10184/org.metabrainz.android D/OpenGLRenderer: Swap behavior 0
2019-02-28 12:31:36.158 10150-10184/org.metabrainz.android D/EGL_emulation: eglCreateContext: 0xe35619a0: maj 3 min 0 rcv 3
2019-02-28 12:31:36.187 10150-10184/org.metabrainz.android D/EGL_emulation: eglMakeCurrent: 0xe35619a0: ver 3 0 (tinfo 0xe3f7edd0)
2019-02-28 12:31:36.414 10150-10184/org.metabrainz.android D/EGL_emulation: eglMakeCurrent: 0xe35619a0: ver 3 0 (tinfo 0xe3f7edd0)
2019-02-28 12:31:44.008 10150-10150/org.metabrainz.android I/AssistStructure: Flattened final assist data: 7032 bytes, containing 1 windows, 28 views
2019-02-28 12:31:46.741 10150-10150/org.metabrainz.android W/ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@8ed160f
2019-02-28 12:31:46.942 10150-10150/org.metabrainz.android D/NetworkSecurityConfig: No Network Security Config specified, using platform default
2019-02-28 12:31:46.944 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setUseSessionTickets(Z)V (light greylist, reflection)
2019-02-28 12:31:46.944 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setHostname(Ljava/lang/String;)V (light greylist, reflection)
2019-02-28 12:31:46.944 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->getAlpnSelectedProtocol()[B (light greylist, reflection)
2019-02-28 12:31:46.944 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setAlpnProtocols([B)V (light greylist, reflection)
2019-02-28 12:31:46.944 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Ldalvik/system/CloseGuard;->get()Ldalvik/system/CloseGuard; (light greylist, reflection)
2019-02-28 12:31:46.944 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Ldalvik/system/CloseGuard;->open(Ljava/lang/String;)V (light greylist, reflection)
2019-02-28 12:31:46.952 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden method Ldalvik/system/CloseGuard;->warnIfOpen()V (light greylist, reflection)
2019-02-28 12:31:47.053 10150-10150/org.metabrainz.android W/abrainz.androi: Accessing hidden field Lsun/misc/Unsafe;->theUnsafe:Lsun/misc/Unsafe; (light greylist, reflection)
2019-02-28 12:31:47.079 10150-10194/org.metabrainz.android D/OkHttp: --> GET http://musicbrainz.org/ws/2/artist/?query=lady%20gaga&limit=100&offset=0
2019-02-28 12:31:47.079 10150-10194/org.metabrainz.android D/OkHttp: --> END GET
2019-02-28 12:31:47.085 10150-10150/org.metabrainz.android W/ViewRootImpl[DashboardActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_ENTER, scanCode=28, metaState=0, flags=0x28, repeatCount=0, eventTime=6988078, downTime=6988050, deviceId=0, source=0x101 }
2019-02-28 12:31:47.159 10150-10194/org.metabrainz.android D/OkHttp: <-- HTTP FAILED: java.net.UnknownServiceException: CLEARTEXT communication to musicbrainz.org not permitted by network security policy
2019-02-28 12:31:47.233 10150-10184/org.metabrainz.android D/EGL_emulation: eglMakeCurrent: 0xe35619a0: ver 3 0 (tinfo 0xe3f7edd0)
2019-02-28 12:31:47.250 10150-10184/org.metabrainz.android D/EGL_emulation: eglMakeCurrent: 0xe35619a0: ver 3 0 (tinfo 0xe3f7edd0)
2019-02-28 12:31:47.257 10150-10184/org.metabrainz.android D/EGL_emulation: eglMakeCurrent: 0xe35619a0: ver 3 0 (tinfo 0xe3f7edd0)
2019-02-28 12:31:47.266 10150-10162/org.metabrainz.android I/abrainz.androi: Background concurrent copying GC freed 11015(831KB) AllocSpace objects, 1(16KB) LOS objects, 49% free, 2MB/4MB, paused 1.228ms total 181.421ms
2019-02-28 12:31:47.086 10150-10150/org.metabrainz.android I/chatty: uid=10085(org.metabrainz.android) identical 4 lines
2019-02-28 12:31:47.086 10150-10150/org.metabrainz.android W/ViewRootImpl[DashboardActivity]: Cancelling event due to no window focus: KeyEvent { action=ACTION_UP, keyCode=KEYCODE_ENTER, scanCode=28, metaState=0, flags=0x28, repeatCount=0, eventTime=6988078, downTime=6988050, deviceId=0, source=0x101 }
```

### Steps to reproduce: 
1. Search for “lady gaga”. Click search
2. New search results activity is shown, but it's empty. Error only shows on console.

Reply:
This seems to be a temporary issue either due to network connection or api search. Because the url 
http://musicbrainz.org/ws/2/artist/?query=lady%20gaga&limit=100&offset=0 replies with a list of results. MusicBrainz search server have been facing problems recently. It may be this reason or a temporary network issue.